src/Process/Job.php

```diff
@@ -26,12 +29,22 @@ protected function _run(): Forked
             $this->_getScheduler()->scheduleStaticJobs();
             $this->_getMaintainer()->updatePendingJobs();
             $this->_getMaintainer()->deleteCompletedJobs();
+            $this->applyMemoryLimit();
             $this->_getForeman()->workWorker();
-        } catch (\Throwable $throwable) {
+        } catch (Throwable $throwable) {
             $this->_getLogger()->critical($throwable->getMessage(), ['exception' => $throwable]);
             $this->_setOrReplaceExitCode(255);
         }
 
         return $this;
     }
+
+    protected function applyMemoryLimit(): JobInterface
+    {
+        $processMaximumMemoryValue = $this->getEnvironmentMemory()->getWorkerMaximumMemoryValue();
+        $this->_getLogger()->debug(sprintf('Applying memory limit of [%s] bytes.', $processMaximumMemoryValue));
+        ini_set('memory_limit', $processMaximumMemoryValue);
```

Comment:
Note that PHP will set `memory_limit` to be 2MiB if it is set to any smaller value. 2MiB is the smallest PHP allows.